cmd/nerdctl/cp.go

```diff
@@ -0,0 +1,229 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/pkg/system"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type cpConfig struct {
+	container  string
+	followLink bool
+	sourcePath string
+	destPath   string
+}
+
+var errFileSpecDoesntMatchFormat = errors.New("filespec must match the canonical format: [container:]file/path")
+
+func newCpCommand() *cobra.Command {
+
+	shortHelp := "Copy files/folders between a container and the local filesystem"
+	longHelp := shortHelp + "\nNOTE: Use '-' as the source to read a tar archive from stdin\n and extract it to a directory destination in a container. \nUse '-' as the destination to stream a tar archive of a\ncontainer source to stdout."
```

Comment:
```
longHelp += `
WARNING: This command is not designed to be used with untrusted containers.
`
```

README.md should contain the same information.

Reply:
Maybe we should also clarify that we won't treat an unexpected behavior of `nerdctl cp` as a vulnerability.
This command almost can't be implemented in a trustable way, and the user must be conscious of that.

(cc @dmcgowan @samuelkarp for security policy)